frontend/src/components/course/course.tsx

```diff
@@ -70,8 +70,15 @@ const Course: FC = () => {
     setChooseMentorModalOpen((prev) => !prev);
   };
 
-  const handleMentorSelectClick = (mentorId: number): void => {
-    dispatch(courseActions.chooseMentor({ id: mentorId }));
+  const handleMentorSelectClick = (
+    mentorId: number,
+    evt: React.MouseEvent,
+  ): void => {
+    dispatch(courseActions.chooseMentor({ id: mentorId }))
+      .unwrap()
+      .then(() => {
+        handleChooseMentorModalToggle(evt);
+      });
```

Comment:
What the user will see in the interval between the request and the server's response?

Reply:
Added pending status to chooseMentor action to show spinner